server.js

```diff
@@ -1,26 +1,78 @@
 var http = require('http');
 var https = require('https');
+var fs = require('fs');
 var path = require('path');
 var express = require('express');
+var helmet = require('helmet');
+var bodyParser = require('body-parser');
+var crypto = require('crypto');
 
 var config = require('./config.server.js');
 
 var buildDir = 'dist';
 
 var app = express();
-var helmet = require('helmet');
+
+var nonceMiddleware = function (req, res, next) {
+  // Cache static html file to avoid reading it from the filesystem on each request
+  if (!global.html) {
+    console.log('Caching static HTML');
+    global.html = fs.readFileSync(staticDir + '/index.html', 'utf8');
+  }
+
+  // Set a unique nonce for each request
+  res.locals.nonce = crypto.randomBytes(16).toString('base64');
+  res.locals.htmlWithNonces = global.html.replace(/<(script)/g, '<$1 nonce="' + res.locals.nonce + '"');
```

Comment:
Recommend using template literals here for readability: 
`'<$1 nonce="' + res.locals.nonce + '"' ` -> `` `<$1 nonce="${res.locals.nonce}"` ``